Metadata/AmazonMetadataBuilder.php

```diff
@@ -114,6 +115,18 @@ protected function getDefaultMetadata()
      */
     protected function getContentType($filename)
     {
+        // NEXT_MAJOR: Remove this BC hack
```

Comment:
note that unless we depend this bundle on guzzle or something else supported by httplug, users will have to explicitly chose their http client implementation. while i think this is the correct way to go, it needs a big note in the setup instructions.